services/wfm/src/main/java/org/openkilda/wfm/topology/event/NetworkTopologyBolt.java

```diff
@@ -0,0 +1,198 @@
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        this.switchService = new SwitchService(transactionManager, repositoryFactory);
+        this.islService = new IslService(transactionManager, repositoryFactory);
+        this.portService = new PortService(transactionManager, repositoryFactory, islCostWhenPortDown);
+        super.prepare(stormConf, context, collector);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void handleInput(Tuple tuple) {
+        String request = tuple.getString(1);
+
+        BaseMessage baseMessage;
+        try {
+            baseMessage = MAPPER.readValue(request, BaseMessage.class);
+        } catch (IOException e) {
+            logger.error("Error during parsing request for Reroute topology", e);
+            return;
+        }
+
+        if (baseMessage instanceof InfoMessage) {
+            InfoMessage message = (InfoMessage) baseMessage;
+            InfoData data = message.getData();
+            Sender sender = new Sender(getOutput(), tuple, message.getCorrelationId());
+
+            if (data instanceof SwitchInfoData) {
+                handleSwitchEvents((SwitchInfoData) data);
+
+            } else if (data instanceof IslInfoData) {
+                handleIslEvents((IslInfoData) data, sender);
+
+            } else if (data instanceof PortInfoData) {
+                handlePortEvents((PortInfoData) data, sender);
+
+            } else {
+                logger.warn("Skip undefined info data type {}", request);
+            }
+        } else {
+            logger.warn("Skip undefined message type {}", request);
+        }
+    }
+
+    private void handleSwitchEvents(SwitchInfoData data) {
+        logger.debug("State update switch {} message {}", data.getSwitchId(), data.getState());
+
+        switch (data.getState()) {
+
+            case ADDED:
+                switchService.createOrUpdateSwitch(SwitchMapper.INSTANCE.map(data));
+                break;
+
+            case ACTIVATED:
+                switchService.activateSwitch(SwitchMapper.INSTANCE.map(data));
+                break;
+
+            case DEACTIVATED:
+            case REMOVED:
+                switchService.deactivateSwitch(SwitchMapper.INSTANCE.map(data));
+                break;
+
+            default:
+                logger.warn("Unknown state update switch info message");
+        }
+    }
+
+    private void handleIslEvents(IslInfoData data, Sender sender) {
+        logger.debug("State update isl {}. Isl state: {}", data.getId(), data.getState());
+
+        switch (data.getState()) {
+            case DISCOVERED:
+                islService.createOrUpdateIsl(IslMapper.INSTANCE.map(data));
+
+                String reason = String.format("Create or update ISL: %s. ISL state: %s",
+                        data.getId(), data.getState());
+                sender.sendRerouteInactiveFlowsMessage(reason);
```

Comment:
in current implementation such feature is a configurable option/toggle in TE. is that toggle obsolete now or you just placed it in another place?

Reply:
We do not support feature toggles. This will be a separate task for the design and implementation of feature toggles.

Reply:
This is part of business logic, it must be on service level, not bolt level.